src/shogun/evaluation/SplittingStrategy.h

```diff
@@ -58,36 +59,32 @@ class CSplittingStrategy: public CSGObject
 	 * @param subset_idx subset index of the to be generated vector indices
 	 * @return newly created vector of subset indices of the specified
 	 * subset is written here.
-	 *
-	 * Error if there are no index sets
 	 */
-	SGVector<index_t> generate_subset_indices(index_t subset_idx);
+	const SGVector<index_t> validation(index_t subset_idx) const;
 
 	/** generates a newly created SGVector<index_t> with inverse indices of the
 	 * subset with the desired index. inverse here means all other indices.
 	 *
 	 * @param subset_idx subset index of the to be generated inverse indices
 	 * @return newly created vector of the subset's inverse indices is
 	 * written here.
-	 *
-	 * Error if there are no index sets
 	 */
-	SGVector<index_t> generate_subset_inverse(index_t subset_idx);
+	const SGVector<index_t> train(index_t subset_idx) const;
 
 	/** @return number of subsets. */
 	index_t get_num_subsets() const;
 
+protected:
```

Comment:
so you make this method private. Why not call it in the constructor then? Then we can make the train/validation method const..

Reply:
I've changed that, thanks.